### PR TITLE
CVE-2004-0488

### PR DESCRIPTION
**Pull request by jake-scali**
Filled out CVE-2004-0488

### CODE REVIEW

File: cves/CVE-2004-0488.yml [hunk 1]
```diff
@@ -31,7 +31,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: This issue with this CVE was a buffer overflow in Apache HTTP Server 2.0.50. By using a user with trusted CA (Certificate Authority) an remote attacker could
```
Comment (kar5254) [inline cves/CVE-2004-0488.yml]:
```suggestion
description: This issue with this CVE was a buffer overflow in Apache HTTP Server 2.0.50. By using a user with trusted CA (Certificate Authority) a remote attacker could
```
sorry if this is a bit grammar nazi tier but good grammar makes it easier to read these types of things

File: cves/CVE-2004-0488.yml [hunk 2]
```diff
@@ -199,4 +202,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: The cause of this error was most likely a coding mistake. Whoever wrote the code did not account for a trusted user to exploit a buffer overflow and this lack of forsight is what led to this vulnerability.
```
Comment (kar5254) [inline cves/CVE-2004-0488.yml]: Saying the error was a "coding mistake" is essentially meaningless. The error was caused by a lack of input sanitization, which is really more of a design flaw than a coding mistake, which would be something like a missing semicolon.

File: cves/CVE-2004-0488.yml (hunk 1 above)
Comment (jake-scali) [reply cves/CVE-2004-0488.yml]: Good find, and has been fixed.

File: cves/CVE-2004-0488.yml (hunk 2 above)
Comment (jake-scali) [reply cves/CVE-2004-0488.yml]: Description has been updated

File: cves/CVE-2004-0488.yml [hunk 3]
```diff
@@ -48,12 +52,12 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
    - commit: d8f71ba6b7c20cad59faa66c1a9bc93f947e651f
-     note:
+     note: modules/ssl/ssl_engine_kernel.c (ssl_hook_UserCheck): Fix buffer
```
Comment (kel-clark) [inline cves/CVE-2004-0488.yml]:
```suggestion
     note: modules/ssl/ssl_engine_kernel.c (ssl_hook_UserCheck) Fix buffer
```
The use of a colon in YAML indicates the start of a new element, and removing this from the line will allow the Travis CI build to pass

File: cves/CVE-2004-0488.yml [hunk 4]
```diff
@@ -199,4 +203,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: The cause of this error was most likely a design flaw. Whoever wrote the code did not account for a trusted user to exploit a buffer overflow and this lack of forsight is what led to this vulnerability.
```
Comment (kel-clark) [inline cves/CVE-2004-0488.yml]:
```suggestion
  answer: The cause of this error was most likely a design flaw. Whoever wrote the code did not account for a trusted user to exploit a buffer overflow and this lack of foresight is what led to this vulnerability.
```

File: cves/CVE-2004-0488.yml [hunk 5]
```diff
@@ -48,12 +52,12 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
    - commit: d8f71ba6b7c20cad59faa66c1a9bc93f947e651f
-     note:
+     note: modules/ssl/ssl_engine_kernel.c (ssl_hook_UserCheck): Fix buffer
 vccs:
   - commit: 1f52b243020ac155bb26ee9b2a9ea960e6eefa06
-    note:
+    note: dropping hungarian notation
```
Comment (kel-clark) [inline cves/CVE-2004-0488.yml]: what does this mean and how did it contribute to creating the vulnerability?

**Review by kel-clark — COMMENTED** [review]
+2 upvotes

File: cves/CVE-2004-0488.yml (hunk 3 above)
Comment (jake-scali) [reply cves/CVE-2004-0488.yml]: Colon has been removed

File: cves/CVE-2004-0488.yml (hunk 4 above)
Comment (jake-scali) [reply cves/CVE-2004-0488.yml]: Spelling has been fixed

File: cves/CVE-2004-0488.yml (hunk 5 above)
Comment (jake-scali) [reply cves/CVE-2004-0488.yml]: Description has been updated

**Pull request merged**